collect_app/src/test/java/org/odk/collect/android/projects/AddProjectDialogTest.kt

```diff
@@ -0,0 +1,39 @@
+package org.odk.collect.android.projects
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.android.material.button.MaterialButton
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.odk.collect.android.R
+import org.odk.collect.android.support.RobolectricHelpers
+
+@RunWith(AndroidJUnit4::class)
+class AddProjectDialogTest {
+
+    @Test
+    fun clickingCancel_dismissesDialog() {
+        val scenario = RobolectricHelpers.launchDialogFragment(AddProjectDialog::class.java)
+        scenario.onFragment { f: AddProjectDialog ->
+            val dialog = f.dialog
```

Comment:
I'd be tempted to do `f.dialog!!` here to catch any problems early (same suggestion for the other test). I feel like I end up using `!!` in tests a lot but completely try to avoid it in non-test code.